docs/middleware.rst

```diff
@@ -0,0 +1,113 @@
+Middlewares
+###########
+
+Middlewares are an essential part of the event sending lifecycle as it's processed
+through them before being sent to the server. Each middleware wraps the next to
```

Comment:
```
Middlewares are an essential part of the event sending lifecycle as they are
processed through each middleware before being sent to the server.
```

Tt's and them are a bit vague so maybe something like above to clarify what is being talked about.